AmplifyPlugins/Core/AWSPluginsCore/Model/Support/SelectionSet.swift

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+/// Contains detail about the current selection set. `pagination` indicates that it contains paginated fields.
+/// `modelField` indicates that the values represent the property names from a `ModelSchema`.
+public enum SelectionSetType {
+    case pagination
+    case modelField
```

Comment:
I don't understand why we need these if we've got decorators that manipulate the selection set